.github/workflows/open-bump-version-pr.yml

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+name: Fork release branch
+
+permissions:
+  # Create release branch and publish release.
+  contents: write
+  # Publish PR.
+  #pull-requests: write
+
+on:
+  workflow_call: {}
+
+jobs:
+  fork-branch:
+    name: Open bump version PR
+    runs-on: ubuntu-latest
+    permissions:
+      # Create release branch and publish release.
+      contents: write
+      # Publish PR.
+      #pull-requests: write
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        path: repo
+
+    - name: Open bump version PR
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        set -x
+
+        git config --global user.name 'github-actions[bot]'
+        git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+
+        pushd ./repo
+
+        # Switch to main branch.
+        git fetch --depth=1 --prune origin main
+        git checkout origin/main
+
+        # Bump version.
+        NEXT_MINOR_VERSION="$(python3 .github/workflows/scripts/bump_minor_version.py)"
+
+        # Commit and push.
+        BUMP_VERSION_BRANCH="workflows/bumpVersion${NEXT_MINOR_VERSION}"
+        git checkout -b "${BUMP_VERSION_BRANCH}"
+
+        git add -A
+        git commit -m "Bump version to v${NEXT_MINOR_VERSION}"
```

Comment:
so what happens when we have major releases? should we have a flag in this pipeline that allow both?

Reply:
For the most part, the exact same logic applies when releasing new major versions. The one exception to this is creating the change to bump the major version. But bumping the major version is easy enough and rare enough, that I don't think it is worth it to create dedicated workflow logic for it.

Reply:
Talked offline